service/prometheus/service.go

```diff
@@ -0,0 +1,82 @@
+package prometheus
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"path"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+)
+
+const (
+	prometheusReloadPath = "/-/reload"
+)
+
+type Prometheus interface {
+	Reload() error
+}
+
+type Config struct {
+	Logger micrologger.Logger
+
+	PrometheusAddress string
+}
+
+func DefaultConfig() Config {
+	return Config{
+		Logger: nil,
+
+		PrometheusAddress: "",
+	}
+}
+
+type Service struct {
+	logger micrologger.Logger
+
+	prometheusAddress string
+}
+
+func New(config Config) (*Service, error) {
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
+	}
+
+	if config.PrometheusAddress == "" {
+		return nil, microerror.Maskf(invalidConfigError, "config.PrometheusAddress must not be empty")
+	}
+
+	u, err := url.ParseRequestURI(config.PrometheusAddress)
+	if err != nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.PrometheusAddress is invalid: %s", err)
+	}
+	u.Path = path.Join(u.Path, prometheusReloadPath)
+
+	service := &Service{
+		logger: config.Logger,
+
+		prometheusAddress: u.String(),
+	}
+
+	return service, nil
+}
+
+func (s *Service) Reload() error {
+	s.logger.Log("debug", fmt.Sprintf("reloading prometheus config: %s", s.prometheusAddress))
+	if _, err := http.Post(s.prometheusAddress, "", nil); err != nil {
```

Comment:
I think we should check the status code and return an error in case it is not implying "success". We should then also have metrics for it in the long run.

Reply:
Yeah, this is good. Done.